src/a11y-dialog.ts

```diff
@@ -307,30 +307,49 @@ function isFocusable(el: HTMLElement) {
  */
 function trapTabKey(el: HTMLElement, event: KeyboardEvent) {
   const focusableChildren = getFocusableChildren(el)
-  const focusedItemIndex = focusableChildren.indexOf(
-    document.activeElement as HTMLElement
-  )
+  const firstFocusableChild = focusableChildren[0]
+  const lastFocusableChild = focusableChildren[focusableChildren.length - 1]
+
+  const activeElement = getDeepActiveElement()
```

Comment:
🔍 We don't need to look through our collection of `focusableChildren` to get an index; we can compare our `activeElement` to the first and last nodes to check if the user is at the beginning or end of the modal.

Reply:
That’s a good point. I’m not sure this was ever a real performance issue but potentially this could be when there are hundreds or thousands of focusable elements. Checking just the first and last is nicer. Well done!